main.tf

```diff
@@ -115,6 +110,57 @@ resource "ibm_iam_authorization_policy" "kms_policy" {
   description                 = "Allow all Event Streams instances in the resource group ${var.resource_group_id} to read from the ${local.kms_service} instance GUID ${var.existing_kms_instance_guid}"
 }
 
+# workaround for https://github.com/IBM-Cloud/terraform-provider-ibm/issues/4478
+resource "time_sleep" "wait_for_kms_authorization_policy" {
+  count      = var.kms_encryption_enabled == false || var.skip_iam_authorization_policy ? 0 : 1
+  depends_on = [ibm_iam_authorization_policy.kms_policy]
+
+  create_duration = "30s"
+}
+
+# Create s2s at service level for provisioning mirroring instance
+resource "ibm_iam_authorization_policy" "es_service_policy" {
```

Comment:
@ocofaigh 
So there are 2 policies required for mirroring-
1- service-to-service (The odd one, which is required for ES mirror instance provisioning.)
2- instance-to-instance (This is to allow source cluster access to target cluster.)

I discussed the same with ES team, for which they mentioned both the policies are required for the mirroring feature.

@srikant-sahu/@JunliWang, can you please pitch in here for service-to-service policy need?

Reply:
the expected workflow for mirroring
- provision source cluster
- create s2s policy at service level (at this point we do not have target cluster CRN yet, so grant at service level)
- provision target cluster (this requires s2s policy because mirrormaker2 on target cluster will be configured to read from source cluster, reader role is needed)
- create s2s policy at instance level (in order to ensure target can read from source cluster, at this point both CRNs are available)
- delete s2s policy at service level (can be done via terraform destroy xxx)

Reply:
@JunliWang The approach you have mentioned around deleting the global auth policy and replacing it with a finer scoped one does not work with terraform.